opencompass/datasets/notdiamond/gpqa.py

```diff
@@ -0,0 +1,81 @@
+import copy
+import csv
+import os
+
+from datasets import Dataset
+
+from opencompass.openicl import BaseEvaluator
+from opencompass.registry import LOAD_DATASET
+
+from .base import BaseDataset
+
+
+@LOAD_DATASET.register_module()
+class GPQADataset(BaseDataset):
+
+    @staticmethod
+    def load(path: str, name: str):
+        cnt = 0
+        data = []
+        data_new = []
+        with open(os.path.join(path, name), "r", encoding="utf-8") as f:
+            reader = csv.reader(f, delimiter=",")
+            for row in reader:
+                if row[7] == "Question":
+                    continue
+                cnt = cnt + 1
+                question = row[7]
+                A = row[8]
+                B = row[9]
+                C = row[10]
+                D = row[11]
+                options = [row[8], row[9], row[10], row[11]]
+                answer = "A"
+
+                data.append(
+                    {
+                        "question": question,
+                        "A": A,
+                        "B": B,
+                        "C": C,
+                        "D": D,
+                        "options": options,
+                        "answer": answer,
+                    }
+                )
+
+                circular_patterns = ["ABCD", "BCDA", "CDAB", "DABC"]  # 更新选项顺序
+                c = circular_patterns[cnt % 4]
+                line = copy.deepcopy(data[cnt - 1])
+                tmp = line["A"]
+                for i in range(4):
+                    line["ABCD"[i]] = line["options"][ord(c[i]) - ord("A")]
+
+                for i in range(4):
+                    if line["ABCD"[i]] == tmp:
+                        line["answer"] = "ABCD"[i]
+                        break
+                data_new.append(line)
+
+        dataset = Dataset.from_list(data_new)
+
+        return dataset
+
+
+class GPQAEvaluator(BaseEvaluator):
+
+    def score(self, predictions, references):
```

Comment:
This should be updated to return per sample accuracy. Ideally renamed as well to avoid confusion

Reply:
Oh, but I'm not using this in `gpqa_config` though. I'm using `NDAccEvaluator`, similar to our Hellaswag eval.

If you'd like, I can have this class inherit from `NDAccEvaluator` and `pass` otherwise. What do you think?

Reply:
Oh I see then no need to change. Thanks